src/unity/python/turicreate/test/test_drawing_classifier.py

```diff
@@ -0,0 +1,265 @@
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+from __future__ import print_function as _
+from __future__ import division as _
+from __future__ import absolute_import as _
+import os as _os
+import turicreate as _tc
+from turicreate.toolkits._main import ToolkitError as _ToolkitError
+import numpy as _np
+from tempfile import mkstemp as _mkstemp
+import coremltools as _coremltools
+from copy import copy as _copy
+import sys as _sys
+from . import util as test_util
+import unittest
+import pytest
+
+def _build_bitmap_data():
+    '''
+    Build an SFrame from 10 saved drawings.
+    '''
+    drawings_dir = _os.path.join(
+        _os.path.dirname(_os.path.realpath(__file__)), 
+        "images", "drawings")
+    sf = _tc.image_analysis.load_images(drawings_dir, with_path=True)
+    sf = sf.rename({"image": "drawing", "path": "label"})
+    sf["label"] = sf["label"].apply(lambda path: path[-10:-5])
+    return sf
+
+def _build_stroke_data():
+    num_rows_in_sframe = 10
+    drawings, labels = [], []
+    random = _np.random.RandomState(100)
+    def _generate_random_point(point = None):
+        if point is not None:
+            dx = random.choice([-1, 0, 1])
+            dy = random.choice([-1, 0, 1])
+            next_x, next_y = point["x"] + dx, point["y"] + dy
+        else:
+            next_x, next_y = random.randint(1000), random.randint(1000)
+        return {"x": next_x, "y": next_y}
+
+    for label in range(num_rows_in_sframe):
+        num_strokes = random.randint(10)
+        drawing = []
+        for stroke_id in range(num_strokes):
+            drawing.append([])
+            num_points = random.randint(500)
+            last_point = None
+            for point_id in range(num_points):
+                last_point = _generate_random_point(last_point)
+                drawing[-1].append(last_point)
+        drawings.append(drawing)
+        labels.append(label)
+
+    return _tc.SFrame({"drawing": drawings, "label": labels})
+
+class DrawingClassifierTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self, pretrained_model_url = None):
+        self.feature = "drawing"
+        self.target = "label"
+        self.check_cross_sf = _build_bitmap_data()
+        self.stroke_sf = _build_stroke_data()
+        self.check_cross_model = _tc.drawing_classifier.create(
+            self.check_cross_sf, 
+            feature = self.feature, 
+            target = self.target, 
+            num_epochs = 10)
+        self.stroke_model = _tc.drawing_classifier.create(
+            self.stroke_sf, 
+            feature = self.feature, 
+            target = self.target, 
+            num_epochs = 1)
+        self.trains = [self.check_cross_sf, self.stroke_sf]
+        self.models = [self.check_cross_model, self.stroke_model]
+
+    def test_create_with_missing_feature(self):
```

Comment:
@nickjong it's true that those two models are ignored in the tests that try to create a model with faulty data. Should I add a `test_create` that creates the two models from `setUpClass` and sets them for future tests? Of course, `pytest` runs tests in alphabetical order so I will have to make sure that this test runs after the `test_create_with_*` and before the tests that need the two models. I personally think it might be okay for the two models to be prepared in `setUpClass` because they are being used in about 14 of 20 tests, but I'm open to suggestions. Let me know!

Reply:
Sorry, I think I was confused, and didn't appreciate that `setUpClass` only runs once total, and its results are shared across individual tests